src/base/net/downloadmanager.cpp

```diff
@@ -512,15 +505,8 @@ namespace
             return;
         }
 
-        auto redirected = static_cast<DownloadHandlerImpl *>(
-                              m_manager->download(Net::DownloadRequest(m_downloadRequest).url(newUrlString)));
-        redirected->m_redirectionCounter = (m_redirectionCounter + 1);
-        connect(redirected, &DownloadHandlerImpl::finished, this, [this](const Net::DownloadResult &result)
-        {
-            m_result = result;
-            m_result.url = url();
-            finish();
-        });
+        ++m_redirectionCounter;
```

Comment:
a bit offtopic, maybe we can use this setter QNetworkRequest::setMaximumRedirectsAllowed instead of manually counting?

Reply:
Oops... I didn't noticed it. Will do.

Reply:
PR updated!